modules/codemod/lib/v7/recategorizeIconButtons.ts

```diff
@@ -0,0 +1,192 @@
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  const requiredImportSpecifiers: string[] = [];
+
+  /**
+   * 1. Find button imports
+   */
+  const {containsCanvasImports, importMap, styledMap} = getImportRenameMap(
+    j,
+    root,
+    '@workday/canvas-kit-react/button'
+  );
+  if (!containsCanvasImports) {
+    console.log(containsCanvasImports);
+
+    return file.source;
+  }
+
+  let buttonType = ''; //default button if no variant is specified
+
+  // Button Mapping
+  // circle -> tertiary
+  // circle-filled -> secondary
+  // inverse -> tertiary inverse
+  // inverse-filled -> secondary inverse
+  // plain/square -> not supported
+  // square-filled -> not supported
+
+  /**
+   * 2. Find `IconButton`
+   *  - If it has no variant, it means it's "circle" which is the default
+   *    - Swap to `TertiaryButton`
+   *    - Add `TertiaryButton` import if it doesn't exist.
+   *  - If not
+   *    - check the variant of the IconButton
+   *    - Swap it out for the appropriate mapping
+   */
+  root
+    .find(
+      j.JSXElement,
+      (value: JSXElement) =>
+        value.openingElement.name.type === 'JSXIdentifier' &&
+        (value.openingElement.name.name === importMap.IconButton ||
+          value.openingElement.name.name === styledMap.IconButton)
+    )
+    .forEach(nodePath => {
+      const attrs = nodePath.value.openingElement.attributes;
+
+      const variantProp = attrs?.find(
+        attr => attr.type === 'JSXAttribute' && attr.name.name === 'variant'
+      );
+      // Default IconButton variant is `circle`
+      if (!variantProp) {
+        buttonType = 'TertiaryButton';
+        updateJSXTag(nodePath, 'TertiaryButton');
+        requiredImportSpecifiers.push('TertiaryButton');
+      } else {
+        const isCircleVariant =
+          ((variantProp as JSXAttribute).value as StringLiteral)?.value === 'circle';
+        const isCircleFilledVariant =
+          ((variantProp as JSXAttribute).value as StringLiteral)?.value === 'circleFilled';
+        const isInverseVariant =
+          ((variantProp as JSXAttribute).value as StringLiteral)?.value === 'inverse';
+        const isInverseFilledVariant =
+          ((variantProp as JSXAttribute).value as StringLiteral)?.value === 'inverseFilled';
+
+        if (isCircleVariant) {
+          nodePath.value.openingElement.attributes?.splice(attrs?.indexOf(variantProp)!, 1);
+          buttonType = 'TertiaryButton';
+        } else if (isCircleFilledVariant) {
+          nodePath.value.openingElement.attributes?.splice(attrs?.indexOf(variantProp)!, 1);
+          buttonType = 'SecondaryButton';
+        } else if (isInverseVariant) {
+          buttonType = 'TertiaryButton';
+        } else if (isInverseFilledVariant) {
+          buttonType = 'SecondaryButton';
+        } else {
+          nodePath.value.openingElement.attributes?.splice(attrs?.indexOf(variantProp)!, 1);
+          buttonType = 'TertiaryButton';
+        }
```

Comment:
```suggestion
        const variantPropValue = ((variantProp as JSXAttribute).value as StringLiteral)?.value;

        buttonType = /filled/gi.test(variantPropValue) ? 'SecondaryButton' : 'TertiaryButton';

        if (!variantPropValue.includes('inverse')) {
          nodePath.value.openingElement.attributes?.splice(attrs?.indexOf(variantProp)!, 1);
        }
```

I guess this big validation can be simplified to above version. It also allow to get rid of lines 83-90

Reply:
nice!!!